### PR TITLE
MINIFICPP-1900 Make the (Python part of the) script extension compile on Windows

### PR DESCRIPTION
**Pull request by fgerlits**
https://issues.apache.org/jira/browse/MINIFICPP-1900

I have tested it with Python only, I'll take a look at Lua scripting later in a separate pull request.

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/script/python/PythonCreator.h [hunk 1]
```diff
@@ -129,7 +129,8 @@ class PythonCreator : public minifi::core::CoreComponent {
       return "";
     }
     const auto python_package_path = std::filesystem::relative(pythonscript, basePath).parent_path();
-    std::vector<std::string> path_elements{python_package_path.begin(), python_package_path.end()};
+    std::vector<std::string> path_elements;
+    std::transform(python_package_path.begin(), python_package_path.end(), std::back_inserter(path_elements), [](const auto& path) { return path.string(); });
```
Comment (szaszm) [inline extensions/script/python/PythonCreator.h:133]:
```suggestion
    std::vector<std::string> path_elements;
    path_elements.reserve(std::distance(python_package_path.begin(), python_package_path.end()));
    std::transform(python_package_path.begin(), python_package_path.end(), std::back_inserter(path_elements), [](const auto& path) { return path.string(); });
```

File: libminifi/include/agent/agent_version.h [hunk 2]
```diff
@@ -18,6 +18,8 @@
 #ifndef LIBMINIFI_INCLUDE_AGENT_AGENT_VERSION_H_
 #define LIBMINIFI_INCLUDE_AGENT_AGENT_VERSION_H_
 
+#undef COMPILER
+
```
Comment (szaszm) [inline libminifi/include/agent/agent_version.h:22]: Why is this needed? Could you add a comment?

Comment (fgerlits) [reply libminifi/include/agent/agent_version.h:22]: TBH, I don't know why it's needed.  Something somewhere `#define`s `COMPILER` to the MSVC version, but I couldn't find what and where.

I'll take another look and add the comment if I find out, but I didn't think `// something somewhere #defines COMPILE to the MSVC version on Windows` would be a useful comment.

Comment (szaszm) [reply libminifi/include/agent/agent_version.h:22]: That's fine, thank you.

Comment (fgerlits) [reply libminifi/include/agent/agent_version.h:25]: I have found the culprit (it's https://github.com/python/cpython/blob/3.10/PC/pyconfig.h) and added a comment.

File: extensions/script/python/PythonCreator.h (hunk 1 above)
Comment (fgerlits) [reply extensions/script/python/PythonCreator.h]: done in ce7b901964cb0c53365c40ba7d5f00fdd701665a

**Review by adam-markovics — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Comment by lordgamez** [comment]
The MacOS I job failure looks to be connected to this change, could you check what caused that failure?

**Review by lordgamez — APPROVED** [review]
